hobo/src/storage.rs

```diff
@@ -151,34 +154,91 @@ impl<Component: 'static> Storage<Component> for SimpleStorage<Component> {
 		self.data.get_mut(&entity).unwrap()
 	}
 }
+// add location make real struct
+pub struct StorageGuard<Component: 'static, Inner: std::ops::Deref<Target = SimpleStorage<Component>>> {
+	pub inner: Option<Inner>,
```

Comment:
this doesn't have to be an `Option` - it is an `Option` in the `*Mut` because I need separate ownership of `Inner` and `World` ref to flush, this can just be an `Inner`